src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs

```diff
@@ -16,8 +16,8 @@ internal class Descriptors
 
         public static DiagnosticDescriptor AZC0002 = new DiagnosticDescriptor(
             nameof(AZC0002),
-            "DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.",
-            "Client method should have cancellationToken as the last optional parameter (both name and it being optional matters)",
+            "DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken or RequestContext parameter called context.",

```

Comment:
I'm not sure we want to be showing about RequestContext in HLC errors yet, lest people think they can use RequestContext there without doing a lot of other work.  It would be preferred to link to an explanation of why to use which, which I don't think we have yet.

Reply:
My experience hitting this was that I got an analyzer error that wasn't clear how to resolve. Most people wouldn't follow the link to read more IMO.  But we could add the context here - https://github.com/Azure/azure-sdk/pull/4752

Reply:
I also think that API reviews should guard against the concern of using RequestContext incorrectly in HLCs.

Reply:
I think @annelo-msft  has a point. We don't want people to be able to pick freely between these two. What if we reword it as "or, in case of protocol methods, RequestContext parameter called context."

Reply:
Can the analyzer detect whether the library is DPG or not?

Reply:
Can we change this to detect protocol method signatures and then test for one or the other?  Or split it into two one that tests for cancellationToken on non protocol and requestContext on protocol?